variable_set_test.go

```diff
@@ -0,0 +1,224 @@
+		// properly passed along.
+		vsl, err := client.VariableSets.List(ctx, orgTest.Name, &VariableSetListOptions{
+			ListOptions: ListOptions{
+				PageNumber: 999,
+				PageSize:   100,
+			},
+		})
+		require.NoError(t, err)
+		assert.Empty(t, vsl.Items)
+		assert.Equal(t, 999, vsl.CurrentPage)
+		assert.Equal(t, 2, vsl.TotalCount)
+	})
+
+	t.Run("when Organization name is invalid ID", func(t *testing.T) {
+		vsl, err := client.VariableSets.List(ctx, badIdentifier, nil)
+		assert.Nil(t, vsl)
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+}
+
+func TestVariableSetsCreate(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	t.Run("with valid options", func(t *testing.T) {
+		options := VariableSetCreateOptions{
+			Name:        String("varset"),
+			Description: String("a variable set"),
+			Global:      Bool(false),
+		}
+
+		vs, err := client.VariableSets.Create(ctx, orgTest.Name, &options)
+		require.NoError(t, err)
+
+		// Get refreshed view from the API
+		refreshed, err := client.VariableSets.Read(ctx, vs.ID, nil)
+		require.NoError(t, err)
+
+		for _, item := range []*VariableSet{
+			vs,
+			refreshed,
+		} {
+			assert.NotEmpty(t, item.ID)
+			assert.Equal(t, *options.Name, item.Name)
+			assert.Equal(t, *options.Description, item.Description)
+			assert.Equal(t, *options.Global, item.Global)
+		}
+	})
+
+	t.Run("when options is missing name", func(t *testing.T) {
+		vs, err := client.VariableSets.Create(ctx, "foo", &VariableSetCreateOptions{
+			Global: Bool(true),
+		})
+		assert.Nil(t, vs)
+		assert.EqualError(t, err, ErrRequiredName.Error())
+	})
+
+	t.Run("when options is missing global flag", func(t *testing.T) {
+		vs, err := client.VariableSets.Create(ctx, "foo", &VariableSetCreateOptions{
+			Name: String("foo"),
+		})
+		assert.Nil(t, vs)
+		assert.EqualError(t, err, ErrRequiredGlobalFlag.Error())
+	})
+}
+
+func TestVariableSetsRead(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	vsTest, vsTestCleanup := createVariableSet(t, client, orgTest, VariableSetCreateOptions{})
+	defer vsTestCleanup()
+
+	t.Run("when the variable set exists", func(t *testing.T) {
+		vs, err := client.VariableSets.Read(ctx, vsTest.ID, nil)
+		require.NoError(t, err)
+		assert.Equal(t, vsTest, vs)
+	})
+
+	t.Run("when variable set does not exist", func(t *testing.T) {
+		vs, err := client.VariableSets.Read(ctx, "nonexisting", nil)
+		assert.Nil(t, vs)
+		assert.Error(t, err)
+	})
+}
+
+func TestVariableSetsUpdate(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	vsTest, _ := createVariableSet(t, client, orgTest, VariableSetCreateOptions{
+		Name:        String("OrinigalName"),
```

Comment:
```suggestion
		Name:        String("OriginalName"),
```

Reply:
I literally struggle to see the problem even with side by side diffs 🤣